Source/DafnyCore/Options/CommonOptionBag.cs

```diff
@@ -297,8 +298,55 @@ public enum TestAssumptionsMode {
           options.DefiniteAssignmentLevel = value ? 1 : 4;
         }
       });
-  }
 
+    DooFile.RegisterLibraryChecks(
+      new Dictionary<Option, DooFile.OptionCheck>() {
+        { UnicodeCharacters, DooFile.CheckOptionMatches },
+        { EnforceDeterminism, DooFile.CheckOptionMatches },
+        // Ideally this feature shouldn't affect separate compilation,
+        // because it's automatically disabled on {:extern} signatures.
+        // Realistically though, we don't have enough strong mechanisms to stop
+        // target language code from referencing compiled internal code,
+        // so to be conservative we flag this as not compatible in general.
+        { OptimizeErasableDatatypeWrapper, DooFile.CheckOptionMatches },
+      }
+    );
+    DooFile.RegisterNoChecksNeeded(
+      Check,
+      Libraries,
+      Output,
+      Plugin,
+      Prelude,
+      Target,
+      Verbose,
+      ErrorLimit,
+      FormatPrint,
+      IsolateAssertions,
+      JsonDiagnostics,
+      QuantifierSyntax,
+      SolverLog,
+      SolverPath,
+      SolverPlugin,
+      SpillTranslation,
+      StdIn,
+      TestAssumptions,
+      WarnShadowing,
+      ManualLemmaInduction,
+      RelaxDefiniteAssignment,
```

Comment:
Yeah and I had the same debate about `EnforceDeterminism`: it's not actually unsound but does feel like the kind of things users will want to apply transitively so I included it. It would be more consistent to include this too, and including an option is a two-way door whereas leaving one out is a one-way door (at least fixing it would likely require rebuilding artifacts, if not an artifact version bump).